src/components/button/config/generate.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright IBM Corp. 2018, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Template configuration generate functions for Button component
+ * @module button/generate
+ */
+import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
+
+// TODO JSDoc
+export function generate(options) {
+  let opts = merge(
+    {
+      element: 'button',
+      content: 'Button',
+      attributes: {},
+      disabled: false,
+      type: 'button',
+      tabIndex: 0,
+      variant: 'primary',
+    },
+    options
+  );
+
+  const selectors = cloneDeep(this.selectors);
+
+  let config = {
+    root: {},
+  };
+
+  Object.keys(selectors.default).forEach(element => {
+    config = merge(config, {
+      [element]: {
+        attributes: {},
+        classNames: selectors.default[element],
+      },
+    });
+  });
+
+  // Element
+
+  config.root.element = opts.element;
+
+  // Content
+
+  config.root.content = opts.content;
```

Comment:
`content` is overloaded in this example, because of the `bx--btn__content` element? I know I broke my rule here of following Carbon React's lead since they don't have that inner span. I called that inner span `bx--btn__content` to show how `content` can be a node's name in selectors object, and still be used alongside `element`, `attributes`, `className` for text.

Does that help? Or is something still misaligned?